File: php/features.php
<?php
declare(strict_types=1);

// GameDevelopment SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GameDevelopmentFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GameDevelopmentBaseFeature();
            case "test":
                return new GameDevelopmentTestFeature();
            default:
                return new GameDevelopmentBaseFeature();
        }
    }
}

File: py/features.py
# GameDevelopment SDK feature factory

from feature.base_feature import GameDevelopmentBaseFeature
from feature.test_feature import GameDevelopmentTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GameDevelopmentBaseFeature(),
        "test": lambda: GameDevelopmentTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
